tracer/src/Datadog.Trace.SourceGenerators/TagsListGenerator/TagListGenerator.cs

```diff
@@ -0,0 +1,327 @@
+﻿// <copyright file="TagListGenerator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Datadog.Trace.SourceGenerators.TagsListGenerator.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Datadog.Trace.SourceGenerators.TagsListGenerator
+{
+    /// <inheritdoc />
+    [Generator]
+    public class TagListGenerator : IIncrementalGenerator
+    {
+        /// <inheritdoc />
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            // Register the attribute source
+            context.RegisterPostInitializationOutput(ctx => ctx.AddSource("TagAttribute.g.cs", Sources.Attributes));
```

Comment:
Are this files commit to git, or always generated at each build? If there not commit, are they already in .gitignore?

Reply:
This proves how little I know about source generators. I thought it created files you check in, like T4 templates.

Reply:
Decided to persist these in the end, as seems good for review purposes apart from anything else. See https://github.com/DataDog/dd-trace-dotnet/pull/2076/commits/400de13ee636f08e845b95d3b48153e6d2ddb1b8